src/main/java/org/reactome/release/cosmicupdate/Main.java

```diff
@@ -0,0 +1,337 @@
+package org.reactome.release.cosmicupdate;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.reactome.release.common.ReleaseStep;
+import org.reactome.release.common.dataretrieval.cosmic.COSMICFileRetriever;
+import org.reactome.util.general.GUnzipCallable;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+public class Main extends ReleaseStep
+{
+	@Parameter(names = {"-u"}, description = "Specifies that the updates should be performed.")
+	private boolean executeUpdate;
+	
+	@Parameter(names = {"-d"}, converter = DurationConverter.class,
+			description = "Download the files, if the age of local files exceeds the specified max age, or if the files don't exist."
+					+ "Omitting this parameter means download will not occur. Specifying -d with a 0 value will force a download."
+					+ "Format for a Duration can be found here: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-"
+					+ " Example: PT48H == \"48 hours\"")
+	private Duration fileAge;
+	
+	@Parameter(names = {"-c"}, description = "The path to the configuration file. Default is src/main/resources/config.properties")
+	private String configPath = "src/main/resources/config.properties";
+	
+//	@Parameter(names = {"-a"}, description = "The path to the authorization file, containing usernames and passwords. Default is src/main/resources/auth.properties")
+//	private static String authPath = "src/main/resources/auth.properties";
+	
+	private static String COSMICMutantExport;
+	private static String COSMICFusionExport;
+	private static String COSMICMutationTracking;
+
+	private static String COSMICMutantExportURL;
+	private static String COSMICFusionExportURL;
+	private static String COSMICMutationTrackingURL;
+	
+	private static String COSMICUsername;
+	private static String COSMICPassword;
+	
+	private static final Logger logger = LogManager.getLogger();
+	private static long personId;
+
+	public static void main(String... args)
```

Comment:
Just curious why using var args instead of a String array here?

Reply:
Why not? It's the same thing.

Reply:
True, but the convention since the beginning of Java has always been to use an explicit array `main(String[] args)`.

It's not a problem, but it jumps out.  If there's no explicit reason for using var args, it might be better to stay with the convention that all Java developers recognize.

Reply:
I've seen `String... args` many times since var-args was first introduced. Some code generators/wizards default to this style, and have been doing so for years.

Personally, I prefer using the newer standard. I also find it more aesthetically pleasing.

Reply:
Okay, I guess I just haven't seen it before.